modules/tinymce/src/core/main/ts/content/InsertContent.ts

```diff
@@ -40,10 +42,25 @@ const processValue = (value: string | DetailsWithContent): ProcessedValue => {
   };
 };
 
+const trimOrPad = (editor: Editor, value: string): string => {
+  const selection = editor.selection;
+  const dom = editor.dom;
+
+  // Check for whitespace before/after value
+  if (/^ | $/.test(value)) {
+    return trimOrPadLeftRight(dom, selection.getRng(), value);
+  } else {
+    return value;
+  }
+};
+
 const insertAtCaret = (editor: Editor, value: string | DetailsWithContent): void => {
-  const result = processValue(value);
+  const { content, details } = processValue(value);
 
-  Rtc.insertContent(editor, result.content, result.details);
+  preProcessSetContent(editor, { content: trimOrPad(editor, content), format: 'html', set: false, selection: true, paste: details.paste }).each((args) => {
+    postProcessSetContent(editor, Rtc.insertContent(editor, args.content, details), args);
```

Comment:
```suggestion
    const insertedContent = Rtc.insertContent(editor, args.content, details);
    postProcessSetContent(editor, insertedContent, args);
```
Maybe? It just looked a little overloaded 🤷